pkg/cli/image/manifest/manifest.go

```diff
@@ -191,6 +191,14 @@ func (o *FilterOptions) Include(d *manifestlist.ManifestDescriptor, hasMultiple
 	return o.OSFilter.MatchString(s)
 }
 
+// IncludeSingle returns true if the provided manifest platform spec is an exact match to FilterByOS
+func (o *FilterOptions) IncludeSingle(d *manifestlist.ManifestDescriptor, hasMultiple bool) bool {
+	if o.FilterByOS == PlatformSpecString(d.Platform) {
+		return true
+	}
+	return false
```

Comment:
Nit:
```
return o.FilterByOS == PlatformSpecString(d.Platform)
```

Reply:
What's the difference between `IncludeSingle` and `Include`?

Reply:
IncludeSingle was taking the regex and pretending it was an exact string - however I removed that and added `^, $` to bookend the filter, but _only_ with `oc image append, extract, and info`, _not_ for `oc imag mirror`